ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java

```diff
@@ -240,20 +240,29 @@ private ValidationResult<TransactionInvalidReason> addTransaction(
       metrics.incrementRejected(isLocal, hasPriority, TRANSACTION_ALREADY_KNOWN, "txpool");
       return ValidationResult.invalid(TRANSACTION_ALREADY_KNOWN);
     }
+    Transaction toAdd = transaction;
+    // if adding a blob tx, and it is missing its blob, is a re-org and we should restore the blob
+    // from cache.
+    if (transaction.getType().supportsBlob() && transaction.getBlobsWithCommitments().isEmpty()) {
+      final Optional<Transaction> maybeCachedBlob = pendingTransactions.restoreBlob(transaction);
+      if (maybeCachedBlob.isPresent()) {
+        toAdd = maybeCachedBlob.get();
```

Comment:
it is more efficient to move this into `reAddTransactions` method, where reorged txs are managed, so avoiding to do this check for every incoming tx.